src/LocalizedApp.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useState } from 'react';
+import { Navigate, useLocation, useNavigate } from 'react-router-dom';
+
+import App from './App';
+import {
+  languageCodes,
+  languageCodesLowerCase,
+  englishCode
+} from './Utilities/LanguageCodes';
+import { localizationObject } from './Utilities/LocalizationLoader';
+import { IntlProvider } from 'react-intl';
+
+const LocalizedApp: React.FC = () => {
+  const [language, setLanguage] = useState(englishCode);
+  const { pathname } = useLocation();
+  const navigate = useNavigate();
+  // constructor(props) {
+  //   super(props);
+  //   this.state = {
+  //     language: getLanguageCode(props.match)
+  //   };
+  //   this.setLanguageCode = this.setLanguageCode.bind(this);
+  // }
+
+  const setLanguageCode = (newLanguage: string, newUrl: string): void => {
+    if (
+      language === newLanguage ||
+      languageCodes.indexOf(newLanguage) < 0
+    ) {
+      return;
+    }
+
+    const urlParts = pathname.split('/'); //TODO check why there are two splits
```

Comment:
Beacuse urlParts is being mutated in one of the if statement branches.

Reply:
Better names for variables Might help

Reply:
I kept it like that